OCR/frontend/src/components/SortableTable/SortableTable.tsx

```diff
@@ -0,0 +1,124 @@
+import {FC, useState} from 'react';
+import {Icon, Table} from "@trussworks/react-uswds";
+import SortArrow from './sort-direction-arrow.svg'
+import SortableIcon from './sortable-icon.svg'
+import './SortableTable.scss'
+
+interface SortableTableProps {
+    data: Array<Map<string, any>>,
+    sortableBy: Array<string> | undefined,
+    defaultSort: string | undefined,
+    defaultDescending: boolean | undefined,
+    columns: Array<string> | undefined,
+    columnNames: Map<string, string> | undefined,
+    formatters: Map<string, (any) => any> | undefined
+}
+
+const SORTS = {
+    '[object Object]': new Intl.Collator(navigator.language).compare,
+    '[object String]': new Intl.Collator(navigator.language).compare,
+    '[object Number]': (a, b) => a - b,
+    '[object Date]': (a, b) => a?.getTime() - b?.getTime()
+}
+
+export const SortableTable: FC<SortableTableProps> = ({
+                                                          data,
+                                                          sortableBy,
+                                                          defaultSort,
+                                                          defaultDescending = false,
+                                                          columns, formatters = {},
```

Comment:
columns = Object.keys(data[0]) you can set this as a default and remove the if statement